lib/Word_Lib/Word_Lemmas.thy

```diff
@@ -622,16 +622,13 @@ lemma shiftr1_0_or_1:"(x::('a::len) word) >> 1 = 0 \<Longrightarrow> x = 0 \<or>
   done
 
 lemma shiftr1_irrelevant_lsb: "bit (x::('a::len) word) 0 \<or> x >> 1 = (x + 1) >> 1"
-  apply (cases \<open>LENGTH('a)\<close>; transfer)
-   apply (simp_all add: take_bit_drop_bit)
-  apply (simp add: drop_bit_take_bit drop_bit_Suc)
-  done
+  by (auto simp add: bit_0 shiftr_def drop_bit_Suc ac_simps elim: evenE)
 
 lemma shiftr1_0_imp_only_lsb:"((x::('a::len) word) + 1) >> 1 = 0 \<Longrightarrow> x = 0 \<or> x + 1 = 0"
   by (metis One_nat_def shiftr1_0_or_1 word_less_1 word_overflow)
 
 lemma shiftr1_irrelevant_lsb': "\<not> (bit (x::('a::len) word) 0) \<Longrightarrow> x >> 1 = (x + 1) >> 1"
-  by (metis shiftr1_irrelevant_lsb)
+  using shiftr1_irrelevant_lsb [of x] by simp
```

Comment:
suggest `using shiftr1_irrelevant_lsb[of x] by simp`
I used to advocate for the `rule [modifier]` style vs `rule[modifier]`, but after some years I came to the realisation that was not such a good idea

Reply:
`Word_Lib` is copied in from the AFP, so outside of breaking compatibility changes I think we would prefer to keep it the same.

I would agree with that change in general however. I prefer modifiers without the whitespace, since for me it makes it easier to take things in at a glance.

Reply:
Currently it'd be Ok to tweak still, I haven't synced our other changes with upstream yet and need to do that anyway. (But yes, all of the Word_Lib changes are verbatim from the AFP)

Reply:
Yeah, ok, let's keep it the same then, I wasn't quite sure what was going on, but if we're pulling from upstream, we're pulling from upstream.